vita3k/app/src/app_init.cpp

```diff
@@ -101,11 +102,86 @@ void update_viewport(EmuEnvState &state) {
     }
 }
 
+void init_paths(Root &root_paths) {
+    const auto dir_sep = std::string{ fs::path::preferred_separator };
+    auto base_path = SDL_GetBasePath();
+    auto pref_path = SDL_GetPrefPath(org_name, app_name);
+    root_paths.set_base_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_pref_path(string_utils::utf_to_wide(pref_path));
+    root_paths.set_log_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_config_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_shared_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_cache_path(fs::path(string_utils::utf_to_wide(base_path)) / "cache" / dir_sep);
+    SDL_free(base_path);
+    SDL_free(pref_path);
+
+#if defined(__linux__) && !defined(__ANDROID__) && !defined(__APPLE__)
+    // XDG Data Dirs.
+    auto env_home = getenv("HOME");
+    auto env_data_dirs = getenv("XDG_DATA_DIRS");
+    auto env_data_home = getenv("XDG_DATA_HOME");
+    auto env_cache_home = getenv("XDG_CACHE_HOME");
+    auto env_config_home = getenv("XDG_CONFIG_HOME");
+
+    if (env_data_home != NULL)
+        root_paths.set_pref_path(fs::path(env_data_home) / app_name / app_name / dir_sep);
+
+    if (env_config_home != NULL)
+        root_paths.set_config_path(fs::path(env_config_home) / app_name / dir_sep);
+    else if (env_home != NULL)
+        root_paths.set_config_path(fs::path(env_home) / ".config" / app_name / dir_sep);
+
+    if (env_cache_home != NULL) {
+        root_paths.set_cache_path(fs::path(env_cache_home) / app_name / dir_sep);
+        root_paths.set_log_path(fs::path(env_cache_home) / app_name / dir_sep);
+    } else if (env_home != NULL) {
+        root_paths.set_cache_path(fs::path(env_home) / ".cache" / app_name / dir_sep);
+        root_paths.set_log_path(fs::path(env_home) / ".cache" / app_name / dir_sep);
+    }
+
+    // Don't assume that base_path is portable.
+    if (fs::exists(root_paths.get_base_path() / "data") && fs::exists(root_paths.get_base_path() / "lang") && fs::exists(root_paths.get_base_path() / "shaders-builtin"))
+        root_paths.set_shared_path(root_paths.get_base_path());
+    else if (env_home != NULL)
+        root_paths.set_shared_path(fs::path(env_home) / ".local/share" / app_name / dir_sep);
+
+    if (env_data_dirs != NULL) {
+        auto env_paths = string_utils::split_string(env_data_dirs, ':');
+        for (auto &i : env_paths) {
+            if (fs::exists(fs::path(i) / app_name)) {
+                root_paths.set_shared_path(fs::path(i) / app_name / dir_sep);
+                break;
+            }
+        }
+    } else if (env_data_home != NULL) {
+        if (fs::exists(fs::path(env_data_home) / app_name / "data") && fs::exists(fs::path(env_data_home) / app_name / "lang") && fs::exists(fs::path(env_data_home) / app_name / "shaders-builtin"))
```

Comment:
Asuming this line, you would only be able to use XDG_DATA_HOME as the place to store these read only files if the y are put there manually because they come in bundled with the zip right?, they dont get generated and have to be put there manually by the user im asuming